### PR TITLE
Constexpr clock_cast

### PR DESCRIPTION
**Pull request by nanoric**
#520

### CODE REVIEW

File: test/clock_cast_test/constexpr.pass.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2017, 2018 Tomasz Kamiński
```
Comment (HowardHinnant) [inline test/clock_cast_test/constexpr.pass.cpp]: You should update the copyright to 2019 and your name.

Comment (nanoric) [reply test/clock_cast_test/constexpr.pass.cpp]: Thanks mentioning.
I am new to contribute to an open-source project. Haven't ever notice that there is a copyright here.

**Comment by HowardHinnant** [comment]
Merged: https://github.com/HowardHinnant/date/commit/66a5aa482e97a9b2ad89d50657a52f94ae492dc6

Sorry it took me so long to review this one.  It was a really good suggestion, but it took me awhile to convince myself that there weren't any downsides.

I made a couple of modifications on merge:

*  All of the added constexpr are C++14 and above.  I could not get a C++11 example working.
*  Some functions that were marked constexpr, I removed the constexpr because they couldn't possibly ever be constexpr.  For example the conversions between system_clock and utc_clock.

**Comment by nanoric** [comment]
Like your rigor.

**Pull request closed without merging**